### PR TITLE
feat: support Latex

### PR DESCRIPTION
**Pull request by ZuzooVn**
# Description

Support Latex response from LLMs

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [x] `optional` I have made corresponding changes to the documentation 
- [x] `optional` I have added tests that prove my fix is effective or that my feature works
- [x] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: web/app/components/base/markdown.tsx [hunk 1]
```diff
@@ -39,6 +39,23 @@ const getCorrectCapitalizationLanguageName = (language: string) => {
 
   return language.charAt(0).toUpperCase() + language.substring(1)
 }
+
+export const preprocessLaTeX = (content: string) => {
```
Comment (crazywoola) [inline web/app/components/base/markdown.tsx]:
```suggestion
const preprocessLaTeX = (content: string) => {
```

File: web/app/components/base/markdown.tsx [hunk 2]
```diff
@@ -82,6 +99,7 @@ const useLazyLoad = (ref: RefObject<Element>): boolean => {
 
 export function Markdown(props: { content: string; className?: string }) {
   const [isSVG, setIsSVG] = useState(false)
+  const latexContent = preprocessLaTeX(props.content)
```
Comment (crazywoola) [inline web/app/components/base/markdown.tsx:90]:
```suggestion
  const latexContent = preprocessLaTeX(content)
```

File: web/app/components/base/markdown.tsx (hunk 1 above)
Comment (crazywoola) [reply web/app/components/base/markdown.tsx]:
```TypeScript
const preprocessLaTeX = (content: string) => 
  content.replace(/\\\[(.*?)\\\]/gs, (_, equation) => `$$${equation}$$`)
         .replace(/\\\((.*?)\\\)/gs, (_, equation) => `$${equation}$`);
```
Guess this is simpler

**Review by crazywoola — CHANGES_REQUESTED** [review]

**Comment by ZuzooVn** [comment]
I updated the code based on the requested change.

**Comment by crazywoola** [comment]
Please fix errors in CI. :)

**Comment by crazywoola** [comment]
Do you have examples that can be used for tests.

**Comment by ZuzooVn** [comment]
> Do you have examples that can be used for tests.

Just ask chatgpt like this: give me some equation example of latex equation

**Review by crazywoola — APPROVED** [review]

**Pull request merged**